sdk/openai/Azure.OpenAI.Inference/tests/OpenAITestBase.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager;
+using Azure.ResourceManager.CognitiveServices;
+using Azure.ResourceManager.CognitiveServices.Models;
+using Azure.ResourceManager.Resources;
+using NUnit.Framework;
+
+namespace Azure.OpenAI.Inference.Tests
+{
+    public abstract class OpenAITestBase : RecordedTestBase<OpenAITestEnvironment>
+    {
+        private const string DeploymentIdVariable = "OPENAI_DEPLOYMENT_ID";
+        private const string EndpointVariable = "OPENAI_ENDPOINT";
+        private const string ResourceGroupName = "openai-test-rg";
+        private const string CognitiveServicesAccountName = "openai-test-account";
+        private const string SubDomain = "sdk";
+        private static AzureLocation Location = AzureLocation.SouthCentralUS;
+        private CognitiveServicesAccountResource _cognitiveServiceAccount;
+        private string _key = "DUMMY_VALUE";
+        private string _endpoint;
+
+        protected OpenAITestBase(bool isAsync, RecordedTestMode? mode = null) : base(isAsync, mode)
+        {
+            HeaderRegexSanitizers.Add(new Core.TestFramework.Models.HeaderRegexSanitizer("api-key", "***********"));
+        }
+
+        private static readonly object _deploymentIdLock = new object();
+        [SetUp]
+        public void CreateDeployment()
+        {
+            if (Mode == RecordedTestMode.Playback)
+            {
+                DeploymentId = Recording.GetVariable(DeploymentIdVariable, null);
+                _endpoint = Recording.GetVariable(EndpointVariable, null);
+            }
+
+            if ((Mode == RecordedTestMode.Live || Mode == RecordedTestMode.Record) && DeploymentId == null)
+            {
+                lock (_deploymentIdLock)
+                {
+                    if (DeploymentId == null)
+                    {
+                        ArmClient armClient = new ArmClient(TestEnvironment.Credential);
+                        var subscription = armClient.GetSubscriptionResource(SubscriptionResource.CreateResourceIdentifier(TestEnvironment.SubscriptionId));
+
+                        ResourceGroupResource resourceGroup = CreateResourceGroupIfNotExists(subscription);
```

Comment:
This does not follow the standard pattern for data plane libraries in which we define a test-resources.json to create the necessary resources for tests. I understand that the "model" cannot be created through test-resources.json, but in such cases where test-specific resources are needed, the pattern is to create those per test while still leveraging test-resources.json for the account-level resources. This ensures that the resources will be created and deleted as part of the test pipeline, even if something were to go wrong in the test run. It also allows for multiple languages to use the same ARM template for creating the test resources. 

If there are gaps in the test-resources script, it would be good to get them documented so that they can be addressed and all libraries + languages can benefit.

/cc @weshaggard @jsquire

Reply:
There are gaps which we can address in the future.  For now to get past the gaps we followed the generated library pattern.

Reply:
I think you are saying that you are following the mgmt library pattern, correct?

Anomaly Detector is a DPG library, but it still uses [test-resources.json](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/anomalydetector/test-resources.json).

Reply:
We should not run live tests until we can follow the data plane live testing model that Josh mentioned. Resource provisioning within the tests themselves often results in problems like leaked resources and quota issues that can be rather hard for the engsys to track down.

Reply:
Resolving the conversation opened this issue for tracking https://github.com/Azure/azure-sdk-for-net/issues/33654